sdks/python/apache_beam/options/pipeline_options_validator_test.py

```diff
@@ -329,7 +329,11 @@ def get_validator(matcher):
     ]
 
     for case in test_case:
-      errors = get_validator(case['on_success_matcher']).validate()
+      matcher = case['on_success_matcher']
+      if matcher and type(matcher) is bytes:
+        errors = get_validator(matcher.decode('utf-8')).validate()
```

Comment:
Are you sure you want to decode the bytes here? Looking at `test_case`, if the matcher type is bytes, then the content is almost surely not decodeable using UTF-8:

```python
>>> import pickle
>>> matcher = pickle.dumps(object)
>>> matcher
b'\x80\x03cbuiltins\nobject\nq\x00.'
>>> matcher.decode("utf-8")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x80 in position 0: invalid start byte
```

Reply:
I think we also need to fix the code that is tested by this test. It looks like we are using a pipeline option  https://github.com/apache/beam/blob/c7d134a35dca488f0cbdf607013a6027599b96ca/sdks/python/apache_beam/options/pipeline_options.py#L671 to pass a pickled piece of code that performs some validation logic in test pipelines. In Python 3 these pickled objects will be returned as bytes, which we will attempt to concat to a string: https://github.com/apache/beam/blob/c7d134a35dca488f0cbdf607013a6027599b96ca/sdks/python/apache_beam/testing/test_pipeline.py#L155 
This test is running into the same problem.

Reply:
@superbobry notice that we are are using a wrapper around pickle: https://github.com/apache/beam/blob/859b8d5c1aade54ea94b1778ea4b889a526e5fa5/sdks/python/apache_beam/internal/pickler.py#L23 so I think decoding using 'utf-8' or even 'latin-1' should be ok.

Reply:
I think I would instead change line 315 to:
```
options.append('%s=%s' % ('--on_success_matcher=' , matcher.decode()))
```
line 155 in test_pipeline.py to: 
```
options.append('--%s=%s' % (k, pickler.dumps(v).decode()))
```
and change `abc` to `b'abc'` in one of the test cases.

What do you all think?

Reply:
Thanks, that looks better and more consistent